problemtools/verifyproblem.py

```diff
@@ -145,7 +154,50 @@ def __str__(self):
     def matches_filter(self, filter_re):
         return filter_re.search(self.strip_path_prefix(self._base)) is not None
 
+    def set_symlinks(self):
+        if not os.path.islink(self.infile):
+            return
+        target = os.path.realpath(self.infile)
+        if target != self.infile and target in self._problem.testcase_by_infile:
+            self.reuse_result_from = self._problem.testcase_by_infile[target]
+
+    def _check_symlinks(self):
+        if not os.path.islink(self.infile):
+            return True
+        nicepath = os.path.relpath(self.infile, self._problem.probdir)
+        in_target = os.path.realpath(self.infile)
+        ans_target = os.path.realpath(self.ansfile)
+        if not in_target.endswith('.in'):
+            self.error("Symbolic link does not point to a .in file for input '%s'" % nicepath)
+            return False
+        if ans_target != in_target[:-3] + '.ans':
+            self.error("Symbolic link '%s' must have a corresponding link for answer file" % nicepath)
+            return False
```

Comment:
Is it conceivable to ever have a situation where one would want to supply a different answer file for the duplicated case?  E.g. one could consider an optimization problem where one gets AC if one beats the answer file produced by some reference solution, and the same case appears several times but with different thresholds of what to beat.  But even if that happens I guess it is so rare that it is not worth the extra complexity, so then one should just duplicate the input file.

Reply:
In that case output validator flags would need to differ, and we emit an error if you use symlinks (precisely because the judge can't just reuse results). One should indeed duplicate the input file in that case.

Reply:
No in the hypothetical example the output validator would do the same thing in both cases (check that output solution is at least as good as answer file solution), only the answer files would differ.  But we anyway agree that this is not a scenario that should be covered so this doesn't matter.